### ISSUE TITLE
bird-rtrlib-cli crashes/stops after few hours

### ISSUE THREAD

**Issue by job** [rtrlib/bird-rtrlib-cli#2]
Hi,

I am running bird-rtrlib-cli on three Ubuntu 14.04 amd64 servers, and on all three the cli client pointed at the IPv4 BIRD instance just stops working. Here is a full log of both STDOUT + STDERR, from start to finish http://paste.ubuntu.com/8024561/

I am running the program like this: ```bird-rtrlib-cli -b /var/run/bird/bird.ctl -r rpki.coloclue.net:8282 2>&1 | tee /tmp/log```

Any suggestions how I can further debugging this?

**Comment by smlng** [comment]
Though, I thought my PR #8 might fix this one, my recent tests showed: it didn't. With Valgrind I wasn't able to reproduce the bahavior, but with GDB I had some _success_, so to speak. Some last lines of output before crashing:

```
bird-rtrlib-cli[23910]: From BIRD: 0000 
bird-rtrlib-cli[23910]: To BIRD: delete roa 58.65.160.0/19 max 19 as 23674

Program received signal SIGPIPE, Broken pipe.
[Switching to Thread 0x7ffff6996700 (LWP 23914)]
0x00007ffff78e688d in write () at ../sysdeps/unix/syscall-template.S:81
81      ../sysdeps/unix/syscall-template.S: Datei oder Verzeichnis nicht gefunden.
```

So we may have to provide a handler for `SIGPIPE` and reconnect, if fired? Any suggestions on that?

**Comment by waehlisch** [comment]
Hmm, why was the socket closed? Anyhow, is probably not our fault. I agree to write handler, which caught the signal and that the program can continue.

**Comment by smlng** [comment]
I don't know why the socket is (temporarily) gone resulting in SIGPIPE, but as restarting `bird-rtrlib-cli` succeeds I assume BIRD handles this somehow and recreates the socket immediately. 

However, [bird-rtrlib-cli.c](https://github.com/rtrlib/bird-rtrlib-cli/blob/master/bird-rtrlib-cli.c#L165) ignores return values (and hence possible errors) on socket read and write. I'll fix that and make a PR.

**Comment by smlng** [comment]
I create a PR #9 which should fix this issue.

I tested the proposed solution for >12h and it worked like charm, no crash whatsoever - communication with BIRD still intact. Though, I saw several SIGPIPEs in syslog, but #9 introduces a SIGPIPE handler and reconnects the bird_socket if needed.

Please review and test - and finally merge #9.

**Comment by waehlisch** [comment]
Fixed in #9
